src/GZCTF/Controllers/GameController.cs

```diff
@@ -897,6 +897,17 @@ public async Task<IActionResult> Submit([FromRoute] int id, [FromRoute] int chal
         if (context.Result is not null)
             return context.Result;
 
+        // Check submission limit if configured for this challenge
+        if (context.Challenge!.SubmissionLimit.HasValue)
+        {
+            var submissionCount = await submissionRepository.GetSubmissionCount(
```

Comment:
Consider adding a submission count field to the challenge instance instrad of counting everytime.